messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -198,16 +198,22 @@ private boolean unsubscribe(String queryName,
                                    interceptAndInvoke(DefaultUnitOfWork.startAndGet(interceptedQuery),
                                                       handler);
                            QueryResponseMessage<R> response = null;
+                           //noinspection unchecked
```

Comment:
This change makes it so that the javadoc of the `QueryBus#scatterGather` method is no longer incorrect, as it states:

> If no handlers are available to provide a result, or when all available handlers throw an exception while attempting to do so, the returned Stream is empty.

With these, exceptions are also reported. Sounds fine to me, but the documentation should be correct on the matter.

Reply:
Additionally, this change will make it so that users of the `DefaultQueryGateway` now experience `IllegalPayloadAccessException` when going through the stream of results. 

This occurs because the `GenericResultMessage#getPayload` will throw this in case of an exception. Is that desired?

Reply:
Good point! I propose to filter out exceptional results on the gateway. @abuijze what's your take on the bus change?

Reply:
Had a discussion with Allard. The conclusion is not to report query handling errors of scatter-gather queries to the calling side. Rationale: scatter-gather is best effort query and it shouldn't care about errors like a simple query.

Reply:
Thanks for the clarification :+1: